src/librbd/internal.cc

```diff
@@ -1576,15 +1576,9 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
   {
     ceph_assert(ceph_mutex_is_locked(ictx->image_lock));
 
-    uint64_t image_size;
-    if (ictx->snap_id == CEPH_NOSNAP) {
-      image_size = ictx->get_image_size(CEPH_NOSNAP);
-    } else {
-      auto snap_info = ictx->get_snap_info(ictx->snap_id);
-      if (snap_info == nullptr) {
-	return -ENOENT;
-      }
-      image_size = snap_info->size;
+    uint64_t image_size = ictx->get_effective_image_size(ictx->snap_id);
+    if (image_size == 0) {
```

Comment:
This is causing the failure in the FSX test suite when it tries to read `0~0`

Reply:
can you point me to the relevant line in the teuthology.log?
can you elaborate on how does it relate to this line?

Reply:
All the failed `fsx` tests:

```
2021-01-08T20:48:05.394 INFO:teuthology.orchestra.run.smithi138.stdout:rbd_read(0, 0) failed
2021-01-08T20:48:05.395 INFO:teuthology.orchestra.run.smithi138.stdout:check_clone: ops->read: No such file or directory
```